packages/flutter/lib/src/rendering/sliver_grid.dart

```diff
@@ -153,12 +153,14 @@ class SliverGridRegularTileLayout extends SliverGridLayout {
     @required this.crossAxisStride,
     @required this.childMainAxisExtent,
     @required this.childCrossAxisExtent,
+    @required this.crossAxisExtent,
```

Comment:
This would be a breaking change since now all users of SliverGridRegularTileLayout need to provide the crossAxisExtent. If we go forward with this, we'd have to follow https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes.

**But before we do that:**
Can you (or @p30arena) explain a little, why this way of fixing the issue is preferred over https://github.com/flutter/flutter/pull/27945, which wouldn't require a breaking API change?

Reply:
It looked more beautiful and clearer to my eyes.
Now that I realize that it has a breaking change, not going to vote for it.

gonna re-open my #27945.

Reply:
@goderbauer this is now explained in the description.

Reply:
> The other PR #27945 tries to calculate the extent from the other provided parameters but cannot guarantee that supplied parameters when recombined always add up to the Cross Axis Extent.

Why is that not guaranteed? The way the parameters are defined should garantee it - otherwise it is a bug in whoever is using this class.

Reply:
I'd argue that the additional parameter actually makes the class harder to use because now there are two ways to get the same information and a user of this class has to make sure that both ways give the same result (otherwise the class will behave funny). I'd prefer to just derive this value from the other values to avoid all that.